src/Core/Schema/Concerns/EagerLoadable.php

```diff
@@ -19,31 +19,51 @@
 
 namespace LaravelJsonApi\Core\Schema\Concerns;
 
+use Closure;
+use InvalidArgumentException;
+
 trait EagerLoadable
 {
 
     /**
-     * @var bool
+     * @var Closure|bool
      */
     private bool $includePath = true;
 
     /**
+     * @param Closure|bool $callback
      * @return $this
      */
-    public function cannotEagerLoad(): self
+    public function canEagerLoad($callback = true): self
     {
-        $this->includePath = false;
+        if (!is_bool($callback) && !$callback instanceof Closure) {
+            throw new InvalidArgumentException('Expecting a boolean or closure.');
+        }
+
+        $this->includePath = $callback;
 
         return $this;
     }
 
+    /**
+     * @return $this
+     */
+    public function cannotEagerLoad(): self
+    {
+        return $this->canEagerLoad(false);
+    }
+
     /**
      * Is the relation allowed as an include path?
      *
      * @return bool
      */
     public function isIncludePath(): bool
     {
-        return $this->includePath;
+        if (is_callable($this->includePath)) {
+            return true === ($this->includePath)();
```

Comment:
I think I'd prefer the value to be thread-cached, i.e.:

```php
if ($this->includePath instanceof \Closure) {
    $this->includePath = ($this->includePath)();
}
```

Reply:
Sure thing!